tests/core/version/test_production_version_migration.py

```diff
@@ -246,7 +245,7 @@ def test_migrate_compatible_version():
         _ScenarioManager._submit(scenario_v2)
 
         assert scenario_v2.d2.read() == 2
-        assert len(_DataManager._get_all(version_number="all")) == 4
+        assert len(_DataManager._get_all(version_number="all")) == 6
```

Comment:
I'm not sure why would the number of data nodes increases here. I will investigate